primer/brand
# Brand - Button accessibility updates

## Summary

Adds accessibility fixes related to [Brand-Button accessibility review](https://github.com/github/primer/issues/1682).

## List of notable changes:

- Added outline on focus when only `aria-disabled` is applied. 
- Added focus indicator and 'disabled' style for Windows High Contrast Mode (WHCM).
- Added use of 'disabled' and 'aria-disabled' separately.
- Hid SVG from screen readers. 

## What should reviewers focus on?

- `aria-disabled` and `disabled` now being separated.
- Styles for button for when `aria-disabled` is applied (focus indication, WHCM support)

## Steps to test:



1. Open up component and use `aria-disabled` attribute control on button. Ensure element is focusable with only `aria-disabled` being applied.
2. Test the same as above, but with the "`disabled`" control, ensure that element is not focusable.
3. See if WHCM styles are applied. 
  (This can be tested within Chrome without Windows through [Chrome Devtools via the "Rendering" tab](https://developer.chrome.com/docs/devtools/rendering/emulate-css/#emulate-css-media-feature-forced-colors). Within the "Emulate CSS media feature forced-colors" section, select the "forced-colors: active" option in the select menu).

## Supporting resources (related issues, external links, etc):

- [Accessibility engineer review for Primer/Brand Button](https://github.com/github/primer/issues/1682)
- [Windows High Contrast Mode & development](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/)

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

<table>
<tr>
<th> Before</th> <th> After </th> <th> Why </th>
</tr>
<tr>
<td valign="top">

![Primer Brand button with "disabled" and "aria-disabled" applied, shows a dimmed button without a focus outline.](https://user-images.githubusercontent.com/26746305/215937632-ac98ff1e-def6-489e-860d-891b24e9e73d.png)

`disabled` and `aria-disabled` applied, no focus indication is present (due to `disabled` attribute, and existing CSS rule).
 </td>


<td valign="top">

![Primer Brand button with only "aria-disabled" applied, shows a dimmed button with a focus outline on focus.](https://user-images.githubusercontent.com/26746305/215937656-96daa9fb-e145-4979-8ac7-7ce66bc0d5a5.png)

Only `aria-disabled` is applied in this example, retains focus indicator due to element still being focusable.
</td>

<td valign="middle">

We'd want to keep the focus outline on the button only when `aria-disabled` is applied. This is because `aria-disabled` will keep the button focusable.

This keeps us in check with [WCAG 2.4.7 Focus visible](https://www.w3.org/WAI/WCAG22/Understanding/focus-Visible.html).
</td>
</tr>

<tr>
<td valign="top">

<img width="170" alt="Primer Brand button within Windows High Contrast Mode (emulated), showing a button with no focus indicator while being actively focused." src="https://user-images.githubusercontent.com/26746305/215944088-052eb1d5-d9b4-42e5-81b7-10720e5be5b0.png">

Focused button (`:focus`) within WHCM.

</td>
<td valign="top">

<img width="170" alt="Primer Brand button within Windows High Contrast Mode (emulated), showing a button with a visible focus indicator while being focused." src="https://user-images.githubusercontent.com/26746305/215944022-48cdfae4-781a-4652-b551-0aa3c1b59bc5.png">

Focused button (`:focus`), within WHCM. Focus outline is present via `outline` CSS rule.

</td>
<td valign="middle">

WHCM by default will strip `box-shadow` from elements visually, meaning that our current focus indicator that uses `box-shadow` needs to be supplemented with an `outline` style. This new rule will only be applied within WHCM (via `@media (forced-colors: active)`).
</td>
</tr>
</table>
